### PR TITLE
Refactor app main 

### PR DESCRIPTION
**Pull request by freddidierRTE**
Nothing in release note

### CODE REVIEW

File: ui/main/src/app/common/utilities.ts [hunk 1]
```diff
@@ -31,4 +33,34 @@ export abstract class Utilities {
         if (obj1 < obj2) return -1;
         return 0;
     }
+
+     // Returns an observable that provides an array. Each item of the array represents even first value of Observable, or it's error
```
Comment (vlo-rte) [inline ui/main/src/app/common/utilities.ts:37]: even -> either
it's -> its

File: ui/main/src/app/modules/core/alert/alert.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!-- Copyright (c) 2018-2020, RTE (http://www.rte-france.com)              -->
```
Comment (vlo-rte) [inline ui/main/src/app/modules/core/alert/alert.component.html:1]: 2018-2020 -> 2022

File: ui/main/src/app/modules/core/application-loading/account-already-used/account-already-used.component.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<!-- Copyright (c) 2018-2020, RTE (http://www.rte-france.com)              -->
```
Comment (vlo-rte) [inline ui/main/src/app/modules/core/application-loading/account-already-used/account-already-used.component.html:1]: 2018-2020 -> 2022

File: ui/main/src/app/modules/core/session-end/session-end.component.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<!-- Copyright (c) 2018-2020, RTE (http://www.rte-france.com)              -->
```
Comment (quinarygio) [inline ui/main/src/app/modules/core/session-end/session-end.component.html:1]: Fix Copyright

File: ui/main/src/app/modules/core/connection-lost/connection-lost.component.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<!-- Copyright (c) 2018-2020, RTE (http://www.rte-france.com)              -->
```
Comment (quinarygio) [inline ui/main/src/app/modules/core/connection-lost/connection-lost.component.html:1]: Fix Copyright

File: ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<!-- Copyright (c) 2018-2020, RTE (http://www.rte-france.com)              -->
```
Comment (quinarygio) [inline ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.html:1]: Fix Copyright

File: ui/main/src/app/modules/core/application-loading/login/login.component.html [hunk 7]
```diff
@@ -24,5 +24,5 @@
             </div>
         </div>
     </form>
-    <div class="alert alert-danger" *ngIf="loginMessage" style="margin:auto;margin-top:50px;width:600px;text-align: center;">
+    <div id="opfab-login-msg" class="alert alert-danger" *ngIf="loginMessage" style="margin:auto;margin-top:50px;width:600px;text-align: center;">
```
Comment (quinarygio) [inline ui/main/src/app/modules/core/application-loading/login/login.component.html:27]: Fix Copyright

File: ui/main/src/app/modules/core/application-loading/app-loaded-in-another-tab/url-lock.service.ts [hunk 8]
```diff
@@ -13,41 +13,33 @@ import {Injectable} from '@angular/core';
...
+    public unlockUrl(): void {
         this.store.setItem(this.IS_OPFAB_URL_CURRENTLY_USED_KEY, 'false');
     }
 
-    public isCurrentUrlAlreadyUsedInBrowser(): boolean {
-        let rawState = this.store.getItem(this.IS_OPFAB_URL_CURRENTLY_USED_KEY);
-        let isConnectedState = !!rawState ? JSON.parse(rawState) : false;
-        return isConnectedState;
+    public isUrlLocked(): boolean {
+        let urlLock = this.store.getItem(this.IS_OPFAB_URL_CURRENTLY_USED_KEY);
```
Comment (vlo-rte) [inline ui/main/src/app/modules/core/application-loading/app-loaded-in-another-tab/url-lock.service.ts:32]: const ?

File: ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+ * This file is part of the OperatorFabric project.
+ */
+
+import {Component, OnInit, TemplateRef, ViewChild} from '@angular/core';
+import {NgbModal, NgbModalRef} from '@ng-bootstrap/ng-bootstrap';
+import {LogOption, OpfabLoggerService} from '@ofServices/logs/opfab-logger.service';
+import {SoundNotificationService} from '@ofServices/sound-notification.service';
+
+// Due to auto-policy in chromium based browsers, if the user does not interact with the application
+// sound is not activated. This component open a modal and by clicking OK the user interacts with the application
```
Comment (vlo-rte) [inline ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts:16]: open -> opens

File: ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+// and activate the sound
```
Comment (vlo-rte) [inline ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts:17]: activate -> activates

File: ui/main/src/app/common/utilities.ts (hunk 1 above)
Comment (freddidierRTE) [reply ui/main/src/app/common/utilities.ts:37]: ok done

File: ui/main/src/app/modules/core/alert/alert.component.html (hunk 2 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/alert/alert.component.html:1]: done

File: ui/main/src/app/modules/core/application-loading/account-already-used/account-already-used.component.html (hunk 3 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/application-loading/account-already-used/account-already-used.component.html:1]: ok

File: ui/main/src/app/modules/core/session-end/session-end.component.html (hunk 4 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/session-end/session-end.component.html:1]: done

File: ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.html (hunk 6 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.html:1]: done

File: ui/main/src/app/modules/core/application-loading/login/login.component.html (hunk 7 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/application-loading/login/login.component.html:27]: done

File: ui/main/src/app/modules/core/application-loading/app-loaded-in-another-tab/url-lock.service.ts (hunk 8 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/application-loading/app-loaded-in-another-tab/url-lock.service.ts:32]: done

File: ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts (hunk 9 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts:16]: done

File: ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts (hunk 10 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/core/application-loading/sound-activation/sound-activation.component.ts:17]: done

**Review by quinarygio — APPROVED** [review]

**Review by vlo-rte — APPROVED** [review]

**Pull request merged**